src/zoom.js

```diff
@@ -15,13 +15,52 @@ export const getShiftedTransform = (
         .translate(labelsWidth + labelsPadding, 0); // put origin at its original position
 };
 
-export default (d3, svg, config, xScale, draw, getEvent) => {
+export function getDomainTransform(d3, config, zoom, domain, xScale, width) {
```

Comment:
please add jsDoc to explain what the function does, together with example input and output